trackintel/io/dataset_reader.py

```diff
@@ -362,3 +362,295 @@ def _calc_overlap_for_candidates(candidates, tpls_this, labels_this, ratio_thres
+    filepath : str
+        path to verification points file within zf.
+    etappen : DataFrame
+        etappen with renamed columns.
+
+    Returns
+    -------
+    pd.DataFrame
+        Filled with aggregated verification points per tripleg.
+    """
+    # we drop border points -> only 6 points
+    num_points = 6
+
+    with zf.open(filepath) as f:
+        vp = pd.read_csv(f, encoding="latin1")
+
+    # insert nan to drop empty rows
+    na_997 = ["{}X", "{}Y", "{}X_CH1903", "{}Y_CH1903"]
+    na_997 = [c.format(f"R{i}_") for i in range(1, num_points + 1) for c in na_997]
+    for col in na_997:
+        vp.loc[vp[col] == -997, col] = np.nan
+
+    # we only keep geometry information
+    cols = [
+        "HHNR",
+        "ETNR",
+        "{}X",
+        "{}Y",
+        "{}X_CH1903",
+        "{}Y_CH1903",
+    ]
+    group = [[c.format(f"R{i}_") for c in cols] for i in range(1, num_points + 1)]
+    rename_to = [c.format("") for c in cols]
+    rename_maps = [{g_col: r_col for (g_col, r_col) in zip(g, rename_to)} for g in group]
+
+    # split by verification point number, rename for later concat, drop emtpy vp
+    gcol = ["X", "Y", "X_CH1903", "Y_CH1903"]
+    vps = [vp[g].rename(columns=rm).dropna(subset=gcol, how="all") for (g, rm) in zip(group, rename_maps)]
+    vp = pd.concat(vps)
+    vp.rename(columns={"HHNR": "user_id"}, inplace=True)  # to be inline with etappen
+    # gather start and end from etappen and put them into the same format
+    sp = etappen[["user_id", "ETNR", "S_X", "S_Y", "S_X_CH1903", "S_Y_CH1903"]].copy()
+    ep = etappen[["user_id", "ETNR", "Z_X", "Z_Y", "Z_X_CH1903", "Z_Y_CH1903"]].copy()
+    sp.rename(columns={"S_X": "X", "S_Y": "Y", "S_X_CH1903": "X_CH1903", "S_Y_CH1903": "Y_CH1903"}, inplace=True)
+    ep.rename(columns={"Z_X": "X", "Z_Y": "Y", "Z_X_CH1903": "X_CH1903", "Z_Y_CH1903": "Y_CH1903"}, inplace=True)
+    vp = pd.concat((sp, vp, ep))  # right order is important
+
+    vp["VP_XY"] = gpd.points_from_xy(x=vp["X"], y=vp["Y"], crs="4236")
+    vp["VP_XY_CH1903"] = gpd.points_from_xy(x=vp["X_CH1903"], y=vp["Y_CH1903"], crs=CRS_CH1903)
+    vp.drop(columns=["X", "Y", "X_CH1903", "Y_CH1903"], inplace=True)
+
+    # aggregate points in the same etappe into a linestring
+    aggfuncs = {"VP_XY": lambda xy: LineString(xy.to_list()), "VP_XY_CH1903": lambda xy: LineString(xy.to_list())}
+    # groupby keeps innner order!
+    vp = vp.groupby(["user_id", "ETNR"], as_index=False).agg(aggfuncs)
+    return vp
+
+
+def _mzmv_to_datetime(col):
+    """Convert time from mzmv to pd.Timestamp on fix date 1970-01-01.
+
+    Parameters
+    ----------
+    col : pd.Series
+        Times stored as strings.
+
+    Returns
+    -------
+    pd.Series
+        Times stored as pd.Timestamp[ns]
+
+    """
+    postfix = [" 1970-01-01"] * len(col)  # no broadcasting possible
+    midnight = col == "24:00:00"
+    col = col.str.cat(postfix)
+    col[midnight] = "00:00:00 1970-01-02"  # 24:00:00 is no valid time
```

Comment:
I changed it to your first suggested change.
We have 854 negative durations but they are negative in any case because they are probably faulty.
![image](https://user-images.githubusercontent.com/49416778/206913087-b9977125-ec42-4d57-8a5c-15e9184321a5.png)